### PR TITLE
Matt Toensing 😎

### PR DESCRIPTION
**Pull request by matttoensing**
I couldn't figure out one condition in a method in the person class. That was tricky! Also, I didn't have enough time to setup the hash in iteration 4 properly. Overall, not the best on my part! Thank you instructors!

### CODE REVIEW

File: lib/event.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  def initialize(name, crafts, attendees)
+    @name = name
+    @crafts = crafts
+    @attendees = attendees
+  end
+
+  def attendee_names
+    @attendees.map do |attendee|
+      attendee.name
+    end
```
Comment (iandouglas) [inline lib/event.rb:14]: nice work here

File: lib/event.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  end
+
+  def craft_with_most_supplies
+    most_supplies = @crafts.max_by do |craft|
+      craft.supplies_required.length
+    end
+    most_supplies.name
```
Comment (iandouglas) [inline lib/event.rb:21]: good job here

File: lib/event.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  end
+
+  def transform_values(collection)
+    collection.map do |element|
+      element.to_s
+    end
```
Comment (iandouglas) [inline lib/event.rb:27]: this works, but this kind of conversion over and over is a little bit 'expensive' from a CPU point of view.

File: lib/event.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  end
+
+  def supply_list
+    supplies_needed = []
+    @crafts.each do |craft|
+      supplies_needed << craft.supplies_required.keys
+    end
+    list = supplies_needed.flatten.uniq
+    transform_values(list)
+  end
```
Comment (iandouglas) [inline lib/event.rb:37]: nice work here

File: lib/event.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+  def attendees_by_craft_interest
+    craft_interests = Hash.new { |hash, key| hash[key] = []}
+    @crafts.each do |craft|
+      craft_interests[craft.name]
+      @attendees.each do |person|
+        craft_interests[craft.name] << person if person.interests.include?(craft.name)
```
Comment (iandouglas) [inline lib/event.rb:44]: nice!!!

File: lib/event.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      end
+    end
+    craft_interests
+  end
+
+  def crafts_that_use(supply)
+    symbol_supply = supply.to_sym
```
Comment (iandouglas) [inline lib/event.rb:51]: nice work changing it to a symbol here

File: lib/event.rb [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    end
+    solution
+  end
+
+  def assign_attendees_to_crafts
+    craft_assignment = Hash.new { |hash, key| hash[key] = []}
+    @crafts.each do |craft|
+      craft_assignment[craft] << @attendees.sample
+    end
+    craft_assignment
```
Comment (iandouglas) [inline lib/event.rb:64]: really well done here

File: lib/person.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    @interests = person_info[:interests]
+    @supplies = {}
+  end
+
+  def add_supply(name, quantity)
+    if @supplies.key?(name)
+      @supplies[name] += quantity
+    else
+      @supplies[name] = quantity
+    end
```
Comment (iandouglas) [inline lib/person.rb:16]: better approach here would be to set line 8 to be `= Hash.new(0)` so every new key is 0 by default, then you would only need line 13 here

File: lib/person.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  # couldn't figure out the second part of the condition statement! Oops!
+  def can_build?(craft)
+    x = transform_values(craft.supplies_required.keys)
+    if @supplies.keys.sort == x.sort
+      true
+    else
+      false
+    end
```
Comment (iandouglas) [inline lib/person.rb:32]: this works ... I might try to use `.all?` here

**Review by iandouglas — COMMENTED** [review]
Fantastic job. Gave you some notes for future reading/learning.

score: pass

**Comment by matttoensing** [comment]
Thank you Ian!

**Pull request closed without merging**